### PR TITLE
Feature/enemies

### PR DESCRIPTION
**Pull request by Davidek776**
This pull request focuses on creating enemy prefabs and enemy patrol script, that give enemies functionality to patrol between points A and B and to as well chase players if it's in the range. In this pull request were also added checkpoints to multiple places in the level.

## TLDR:

---

1. Add enemies prefabs
2. Add EnemyPatrol script
3. Update the Respawn script (So enemies can also kill)
4. Checkpoints composition

### CODE REVIEW

File: Lost Together/Assets/Scripts/EnemyPatrol.cs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    {
+        rb = GetComponent<Rigidbody2D>();
+        anim = GetComponent<Animator>();
+        currentPoint = pointB; 
+        anim.SetBool("isRunning", true);
+    }
+
+   private void Update()
+{
+    GameObject[] players = GameObject.FindGameObjectsWithTag("Player1");
```
Comment (simonfrayer) [inline Lost Together/Assets/Scripts/EnemyPatrol.cs]: Line 26 and 27, why do you assign those as GameObject arrays? We know we always have only one Player1 and Player2 object in the scene so I think you can refactor this.

File: Lost Together/Assets/Scripts/MovementController.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public class MovementController : MonoBehaviour
     public Transform groundCheck;
 
     private Animator animator;
-    
+
```
Comment (simonfrayer) [inline Lost Together/Assets/Scripts/MovementController.cs:19]: I guess delete this empty line?

**Review by simonfrayer — COMMENTED** [review]
2 comments otherwise looks good, also I think in the scene it would be good ideal to make the pointA and pointB for the enemy movement range as child component of the enemy prefab so that it can be easily managed.

**Comment by Davidek776** [comment]
I've fixed the issues. However, it's not possible to make points as a child of the enemy, because then they would have a position relative to their parent, so they would move with the enemy. So it won't be working.

**Comment by simonfrayer** [comment]
> I've fixed the issues. However, it's not possible to make points as a child of the enemy, because then they would have a position relative to their parent, so they would move with the enemy. So it won't be working.

Couldn't we make it at least a prefab?

**Comment by Davidek776** [comment]
Ok, I made a prefab from it.

**Pull request merged**